### ISSUE TITLE
Add authentication module

### ISSUE THREAD

**Issue by ryanw-se** [SunsetAPI/java-wrapper#7]
Should be pretty straight forward, login should only be dependent for manipulating account details like your friends list, biography, networking handles, etc. Need to spend time verifying that the end result is secure and not dependent in order for the other parts of the API to function.

**Comment by ryanw-se** [comment]
This blocks full functionality of issue #8 which is restricted to only 16 results until they authentication and friends can be retrieved from the friendship dashboard.

**Comment by ryanw-se** [comment]
This blocks full functionality of issue #15 which is restricted to just viewing basic statistics and information about the team, no actions such as removing players from the team will be able to be performed until authentication has landed.

**Comment by ryanw-se** [comment]
I will not be actively working on an authentication module, not much research was put into this and there really isn't any way to securely store credentials, you can see the Overcast Network's official post here, https://oc.tc/forums/topics/55ccf43dc29635d931007621 for more information.

**Comment by twizmwazin** [comment]
If this is used locally, passwords can be secure. If website communication is managed via https and their is no middleman server, it can be safely assumed that the passwords will not be stolen, as they could be stolen in the same way as a user submitting their username and password directly on the website.

**Comment by ryanw-se** [comment]
@twizmwazin Just going to explain the method behind the madness, on why I made the decision I did, when I did and just to clarify a few things. This project from the start was intended to be a statistics API, reading stats from various parts of the website to replace the string reading in the Undercast mod.

That of course doesn't require authentication, the only things at the moment that require authentication is managing teams and getting more than 16 friends. Friends can be read from the /friends list on the OCN server and my API was really only intended to get information, not PUT/POST it.

All of these projects popping up, like Sunshine and others requiring authentication really desensitizes the community and they get into a habit of being care free with their credentials. This is an open source project and although I can't control what people do, I can sure as hell discourage it.

Adding an authentication module just makes it easier for malicious users to take advantage of care free users who think everything is okay and enter their account information, regardless of whether its hosted locally or not.
